src/os/bluestore/NVMEDevice.cc

```diff
@@ -99,7 +101,7 @@ class SharedDriverData {
         ctrlr(c),
         ns(ns_) {
     sector_size = spdk_nvme_ns_get_sector_size(ns);
-    block_size = std::max(CEPH_PAGE_SIZE, sector_size);
+    block_size = std::max<uint64_t>(CEPH_NVME_BLK_SIZE, sector_size);
```

Comment:
probably we can just use the `spdk_nvme_ns_get_extended_sector_size(ns)` for block size, because in this case we are performing i/o on an NVMe device, the block size is sector + meta data.

Reply:
see also https://github.com/spdk/spdk/blob/3e2297140ca17ff97bda235b2b5962ebf185953f/lib/bdev/nvme/bdev_nvme.c#L774

Reply:
@tchaikov yes, we can assign block_size with the spdk_nvme_ns_get_extended_sector_size(ns), and remove sector_size parameter. Thanks for the comments. I will update the change soon.